modules/text_server_adv/thorvg_svg_in_ot.cpp

```diff
@@ -74,6 +74,53 @@ void tvg_svg_in_ot_free(FT_Pointer *p_state) {
 	memdelete(state);
 }
 
+static void construct_xml(Ref<XMLParser> &parser, double &r_embox_x, double &r_embox_y, String *p_xml, int64_t &r_tag_count) {
+	if (parser->get_node_type() == XMLParser::NODE_ELEMENT && parser->get_node_name() == "svg") {
+		if (parser->has_attribute("viewBox")) {
+			PackedStringArray vb = parser->get_named_attribute_value("viewBox").split(" ");
+
+			if (vb.size() == 4) {
+				r_embox_x = vb[2].to_float();
+				r_embox_y = vb[3].to_float();
+			}
+		}
+		if (parser->has_attribute("width")) {
+			r_embox_x = parser->get_named_attribute_value("width").to_float();
+		}
+		if (parser->has_attribute("height")) {
+			r_embox_y = parser->get_named_attribute_value("height").to_float();
+		}
+	}
+	if (parser->get_node_type() == XMLParser::NODE_ELEMENT) {
+		*p_xml += vformat("<%s", parser->get_node_name());
+
+		bool is_svg_tag = parser->get_node_name() == "svg";
```

Comment:
There is some small repetitions here, could be  : 
```
if (parser->get_node_type() == XMLParser::NODE_ELEMENT)
{
    *p_xml += vformat("<%s", parser->get_node_name());
    bool is_svg_tag = parser->get_node_name() == "svg";
    for (int i = 0; i < parser->get_attribute_count(); i++) {
		String aname = parser->get_attribute_name(i);
	        if (is_svg_tag && (aname == "viewBox")) {
			PackedStringArray vb = parser->get_named_attribute_value("viewBox").split(" ");

			if (vb.size() == 4) {
				r_embox_x = vb[2].to_float();
				r_embox_y = vb[3].to_float();
			  }
		}
              else if (is_svg_tag && (aname == "width")) {
                        r_embox_x = parser->get_named_attribute_value("width").to_float();
              } else if (is_svg_tag && (aname == "height")) {
                       r_embox_y = parser->get_named_attribute_value("height").to_float();
               } else {
		     *p_xml += vformat(" %s=\"%s\"", aname, parser->get_attribute_value(i));
               }
    }
```  

And as you have the index, you could probably do better than `get_named_attribute_value`